libs/community/langchain_community/llms/solar.py

```diff
@@ -0,0 +1,107 @@
+from typing import Any, Dict, List, Optional
+
+import requests
+from langchain_core.callbacks import CallbackManagerForLLMRun
+from langchain_core.language_models import LLM
+from langchain_core.pydantic_v1 import BaseModel, Field, SecretStr, root_validator
+from langchain_core.utils import convert_to_secret_str, get_from_dict_or_env
+
+from langchain_community.llms.utils import enforce_stop_tokens
+
+SOLAR_SERVICE_URL_BASE = "https://api.upstage.ai/v1/solar"
+
+
+class _SolarClient(BaseModel):
+    """An API client that talks to the Solar server."""
+
+    api_key: SecretStr
+    """The API key to use for authentication."""
+    base_url: str = SOLAR_SERVICE_URL_BASE
+
+    def completion(self, request: Any) -> Any:
+        headers = {"Authorization": f"Bearer {self.api_key.get_secret_value()}"}
+        response = requests.post(
+            f"{self.base_url}/chat/completions",
+            headers=headers,
+            json=request,
+        )
+        if not response.ok:
+            raise ValueError(f"HTTP {response.status_code} error: {response.text}")
+        return response.json()["choices"][0]["message"]["content"]
+
+
+class SolarCommon(BaseModel):
+    _client: _SolarClient
+    base_url: str = SOLAR_SERVICE_URL_BASE
+    solar_api_key: Optional[SecretStr] = Field(default=None, alias="api_key")
+    """Solar API key. Get it here: https://console.upstage.ai/services/solar"""
+    model_name: str = Field(default="solar-1-mini-chat", alias="model")
+    """Model name. Available models listed here: https://console.upstage.ai/services/solar"""
+    max_tokens = 1024
+    temperature = 0.3
+
+    class Config:
+        allow_population_by_field_name = True
+
+    @property
+    def lc_secrets(self) -> dict:
+        return {"solar_api_key": "SOLAR_API_KEY"}
+
+    @property
+    def _default_params(self) -> Dict[str, Any]:
+        return {
+            "model": self.model_name,
+            "max_tokens": self.max_tokens,
+            "temperature": self.temperature,
+        }
+
+    @property
+    def _invocation_params(self) -> Dict[str, Any]:
+        return {**{"model": self.model_name}, **self._default_params}
+
+    @root_validator(pre=True)
+    def build_extra(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        return values
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
```

Comment:
I assumed this function would raise an exception if it detected any issues in its environment. Did I overlook something?